### ISSUE TITLE
Optional list should not require the field to be set in the frontmatter

### ISSUE THREAD

**Issue by timlrx** [contentlayerdev/contentlayer#95]
The following document type:
```
export const Blog = defineDocumentType(() => ({
  name: 'Blog',
  filePathPattern: 'blog/*.mdx',
  bodyType: 'mdx',
  fields: {
    title: { type: 'string', required: true },
    date: { type: 'date', required: true },
    tags: { type: 'list', of: { type: 'string' },
    lastmod: { type: 'date' },
  },
}))
```
and mdx file frontmatter:
```
title: My blog post
date: '2020-12-21'
```
gives this error message:
![image](https://user-images.githubusercontent.com/28362229/148639794-d86df4ee-38e5-47b6-abc3-d8a741805193.png)

It would be better if an optional list does not require the field to be set in the frontmatter, similar to how lastmod can be omitted without errors.

**Comment by rsipakov** [comment]
As a temporary solution, if there is no need to specify tags. Does it work?

```
title: My blog post
date: '2020-12-21'
tags: []
```

**Comment by timlrx** [comment]
Yes, that works but would prefer a fix so I do not need to edit my old markdown files.

**Comment by schickling** [comment]
This should be fixed in with version [0.0.34](https://github.com/contentlayerdev/contentlayer/releases/tag/v0.0.34).
